libraries/freertos_plus/standard/freertos_plus_tcp/source/portable/NetworkInterface/STM32Fxx/NetworkInterface.c

```diff
@@ -427,33 +372,47 @@ BaseType_t xResult;
 
 	if( xEMACTaskHandle == NULL )
 	{
-		#if( ipconfigZERO_COPY_TX_DRIVER != 0 )
+		if( xTXDescriptorSemaphore == NULL )
 		{
-			if( xTXDescriptorSemaphore == NULL )
-			{
-				xTXDescriptorSemaphore = xSemaphoreCreateCounting( ( UBaseType_t ) ETH_TXBUFNB, ( UBaseType_t ) ETH_TXBUFNB );
-				configASSERT( xTXDescriptorSemaphore );
-			}
+			xTXDescriptorSemaphore = xSemaphoreCreateCounting( ( UBaseType_t ) ETH_TXBUFNB, ( UBaseType_t ) ETH_TXBUFNB );
+			configASSERT( xTXDescriptorSemaphore );
```

Comment:
Should this be a runtime check rather than an assert? I'm wondering what will happen here in production code, following resource exhaustion, with assert compiled out.